crypto/src/lib.rs

```diff
@@ -403,6 +328,25 @@ impl KeyPair {
     }
 }
 
+#[cfg(feature = "std")]
```

Comment:
This is an interesting approach. The most readable `impl<'de> Deserialize<'de>` which wasn't redundant. I wish we used this approach more.